storage.go

```diff
@@ -3,58 +3,137 @@ package certmagicsql
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
 	"io/fs"
+	"os"
 	"time"
 
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
 	"github.com/caddyserver/certmagic"
 	_ "github.com/lib/pq"
+	"go.uber.org/zap"
 )
 
 var (
 	_ certmagic.Storage = (*PostgresStorage)(nil)
 )
 
-// DB represents the required database API. You can use a *database/sql.DB.
-type DB interface {
-	BeginTx(context.Context, *sql.TxOptions) (*sql.Tx, error)
-	ExecContext(context.Context, string, ...interface{}) (sql.Result, error)
-	QueryRowContext(context.Context, string, ...interface{}) *sql.Row
-	QueryContext(context.Context, string, ...interface{}) (*sql.Rows, error)
+type PostgresStorage struct {
+	logger *zap.Logger
+
+	QueryTimeout time.Duration `json:"query_timeout,omitempty"`
+	LockTimeout  time.Duration `json:"lock_timeout,omitempty"`
+	Database     *sql.DB       `json:"-"`
+	Host         string        `json:"host"`
```

Comment:
@gabrielmocan It looks like this change removes the `connection_string` config field. Can we have it so both are possible? I.e. allow users to provide a connection string or have the connection fields separated as you've designed it?

Reply:
Ok, I have a fix for that. Will submit in single PR.